qiskit/transpiler/preset_passmanagers/level3.py

```diff
@@ -108,17 +114,16 @@ def _direction_condition(property_set):
 
     _direction = [CXDirection(coupling_map)]
 
-    # 5. 1q rotation merge and commutative cancellation iteratively until no more change in depth
+    # 6. 1q rotation merge and commutative cancellation iteratively until no more change in depth
     _depth_check = [Depth(), FixedPoint('depth')]
 
     def _opt_control(property_set):
         return not property_set['depth_fixed_point']
 
-    _opt = [RemoveResetInZeroState(),
-            Collect2qBlocks(), ConsolidateBlocks(),
+    _opt = [Collect2qBlocks(), ConsolidateBlocks(),
             Unroller(basis_gates),  # unroll unitaries
             Optimize1qGates(), CommutativeCancellation(),
-            OptimizeSwapBeforeMeasure(), RemoveDiagonalGatesBeforeMeasure()]
+            OptimizeSwapBeforeMeasure()]
```

Comment:
Thinking about this a little further, I think we still want `RemoveResetInZeroState` and `RemoveDiagonalGatesBeforeMeasure` here (in addition to where they were added), just to clean up any gates that may have been introduced during the other optimization passes.

Reply:
Wasn't the original issue about how these passes don't have any effect across barriers? Given that this stage occurs after mapping, barriers would already have been inserted (unless we are in the "no coupling map" regime). 

Basically I'm worried about invoking useless passes that slow down the process for 90% of cases, while providing tiny benefits to 10% of cases. Given that this is level 3 I guess timing is not too critical, but just something to keep in mind. I'm fine whichever way you decide.